packages/vue/src/auth/mapAuthState.ts

```diff
@@ -53,44 +61,132 @@ export const connectMapState = (_this: any) => {
+export const mapMfaActions = <K extends keyof MfaActions>(action: K): MfaActions[K] => actionGetter(action);
+export const mapProfileActions = <K extends keyof ProfileActions>(action: K): ProfileActions[K] => actionGetter(action);
+export const mapSecurityPolicyActions = <K extends keyof SecurityPolicyActions>(action: K): SecurityPolicyActions[K] =>
+  actionGetter(action);
+export const mapSignupActions = <K extends keyof SignUpActions>(action: K): SignUpActions[K] => actionGetter(action);
+export const mapSocialLoginActions = <K extends keyof SocialLoginActions>(action: K): SocialLoginActions[K] =>
+  actionGetter(action);
+export const mapSsoActions = <K extends keyof SSOActions>(action: K): SSOActions[K] => actionGetter(action);
+export const mapTeamActions = <K extends keyof TeamActions>(action: K): TeamActions[K] => actionGetter(action);
+export const mapTenantsActions = <K extends keyof TenantsActions>(action: K): TenantsActions[K] => actionGetter(action);
 
+export const connectFronteggStoreV3 = (store: EnhancedStore) => {
+  const initialState = store.getState();
 
-const actionGetter = (action: any) => (...args: any) => (ActionsHolder.getAction(action) as any)(...args);
+  const authState = reactive({ ...initialState.auth });
+
+  const unsubscribe = store.subscribe(() => {
+    const state = store.getState().auth;
+
+    Object.entries(state).forEach(([key, value]) => {
+      authState[key as keyof AuthState] = value;
+    });
+  });
+
+  return { authState, unsubscribe };
+};
+
+export const useFronteggLoaded = () => {
+  const fronteggLoaded = inject('fronteggLoaded') as boolean;
+
+  return fronteggLoaded;
+};
+
+export const useUnsubscribeFronteggStore = () => {
+  const unsubscribeFronteggStore = inject('unsubscribeFronteggStore') as () => void;
+
+  return unsubscribeFronteggStore;
+};
+
+export const useAuthState = () => {
+  const authState = inject('authState') as AuthState;
 
+  return authState;
+};
+
+export const useFronteggStore = () => {
+  const fronteggStore = inject('fronteggStore');
+  return fronteggStore;
+};
+
+export const useFrontegg = () => {
+  const fronteggLoaded = useFronteggLoaded();
+  const unsubscribeFronteggStore = useUnsubscribeFronteggStore();
+  const authState = useAuthState();
+  const fronteggAuth = inject('fronteggAuth') as FronteggAuthService;
+
+  const fronteggStore = useFronteggStore() as EnhancedStore;
 
-type MapLoginActions = LoginActions & {'loginWithRedirect': LoginActions['requestHostedLoginAuthorize']};
-export const mapAuthActions =
-  <K extends keyof AuthActions>(action: K): AuthActions[K] => actionGetter(action);
-export const mapLoginActions =
-  <K extends keyof MapLoginActions>(action: K): MapLoginActions[K] => {
-    if (action === 'loginWithRedirect') {
-      return actionGetter('requestHostedLoginAuthorize');
+  const loginWithRedirect = () => {
+    // @ts-ignore
+    if (!fronteggAuth.router?.currentRoute.path.startsWith(authState.routes.hostedLoginRedirectUrl)) {
```

Comment:
use destructuring

Reply:
we cant destructure reactive prop

Reply:
so can be done like this to improve readability
```
const currentRoute = fronteggAuth.router?.currentRoute.path
const hostedLoginRedirectUrl = authState.routes.hostedLoginRedirectUrl
if (!currentRoute.startsWith(hostedLoginRedirectUrl)) 
```